app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_post_counter
+    user.update(posts_count: user.posts.count)
+  end
```

Comment:
## [OPTIONAL]

- [ ] I think it would be nce to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

```suggestion
  # app/models/post.rb
  after_save :update_post_counter

  private
  def update_post_counter
    user.update(posts_count: user.posts.count)
  end
```